bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/ZoneHandler.java

```diff
@@ -0,0 +1,194 @@
+    private final Logger logger = LoggerFactory.getLogger(ZoneHandler.class);
+    public @Nullable String number;
+
+    public ZoneHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing zone handler.");
+        final OmnilinkBridgeHandler bridgeHandler = super.getOmnilinkBridgeHandler();
+
+        super.initialize();
+        if (bridgeHandler != null) {
+            updateZoneProperties(bridgeHandler);
+        } else {
+            logger.debug("Could not connect to Bridge, failed to update Zone properties!");
+        }
+    }
+
+    private void updateZoneProperties(OmnilinkBridgeHandler bridgeHandler) {
+        final List<AreaProperties> areas = super.getAreaProperties();
+        if (areas != null) {
+            for (AreaProperties areaProperties : areas) {
+                int areaFilter = super.bitFilterForArea(areaProperties);
+
+                ObjectPropertyRequest<ZoneProperties> objectPropertyRequest = ObjectPropertyRequest
+                        .builder(bridgeHandler, ObjectPropertyRequests.ZONE, getThingNumber(), 0).selectNamed()
+                        .areaFilter(areaFilter).build();
+
+                for (ZoneProperties zoneProperties : objectPropertyRequest) {
+                    if (zoneProperties.getZoneType() <= SENSOR_TYPE_PROGRAMMABLE_ENERGY_SAVER_MODULE) {
+                        Map<String, String> properties = editProperties();
+                        properties.put(THING_PROPERTIES_NAME, zoneProperties.getName());
+                        properties.put(THING_PROPERTIES_AREA, Integer.toString(areaProperties.getNumber()));
+                        updateProperties(properties);
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand called for channel: {}, command: {}", channelUID.getAsString(), command);
+        int mode;
+
+        if (!(command instanceof StringType)) {
+            logger.warn("Invalid command: {}, must be StringType", command);
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case CHANNEL_ZONE_BYPASS:
+                mode = OmniLinkCmd.CMD_SECURITY_BYPASS_ZONE.getNumber();
+                break;
+            case CHANNEL_ZONE_RESTORE:
+                mode = OmniLinkCmd.CMD_SECURITY_RESTORE_ZONE.getNumber();
+                break;
+            default:
+                mode = -1;
+        }
+        int zoneNumber = getThingNumber();
+        int areaNumber = getAreaNumber();
+        logger.debug("mode {} on zone {} with code {}", mode, zoneNumber, command.toFullString());
+        char[] code = command.toFullString().toCharArray();
+        if (code.length != 4) {
+            logger.error("Invalid code length, code must be 4 digits");
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be warn. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging